dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessDefinitionMapper.xml

```diff
@@ -208,4 +208,17 @@
         SELECT DISTINCT(id) as project_id
         FROM t_ds_project
     </select>
+
+    <select id="queryDefinitionCodeListByProjectCodes" resultType="java.lang.Long">
+        select
+        code
+        from t_ds_process_definition
+        where 1=1
+        <if test="projectCodes != null and projectCodes.size() != 0">
+            and project_code in
+            <foreach collection="projectCodes" index="index" item="i" open="(" separator="," close=")">
+                #{i}
+            </foreach>
+        </if>
```

Comment:
If the `projectCodes` is a required param, we would better not check it is empty here, if someone call this method but forget to set the projectCodes, it will get all workflow definition codes.
```suggestion
        where project_code in
            <foreach collection="projectCodes" index="index" item="i" open="(" separator="," close=")">
                #{i}
            </foreach>
```

Reply:
No, User just can see the project which authed